pkg/proxy/logger/endpoint/epinfo.go

```diff
@@ -40,42 +40,45 @@ func NewEndpointInfoRegistry(ipc *ipcache.IPCache, endpointManager endpointmanag
 	}
 }
 
-func (r *endpointInfoRegistry) FillEndpointInfo(info *accesslog.EndpointInfo, addr netip.Addr, id identity.NumericIdentity) {
+func (r *endpointInfoRegistry) FillEndpointInfo(info *accesslog.EndpointInfo, addr netip.Addr) {
 	var ep *endpoint.Endpoint
 	if addr.IsValid() {
 		if addr.Is4() {
 			info.IPv4 = addr.String()
 		} else {
 			info.IPv6 = addr.String()
 		}
-
-		// Get (local) endpoint identifier to be reported by cilium monitor
-		ep = r.endpointManager.LookupIP(addr)
-		if ep != nil {
-			info.ID = ep.GetID()
-		}
 	}
 
 	// Only resolve the security identity if not passed in, as it may have changed since
 	// reported by the proxy. This way we log the security identity and labels used for
 	// policy enforcement, if any.
-	if id == 0 {
+	if info.ID == 0 || info.Identity == 0 {
+		// Get (local) endpoint identifier to be reported by cilium monitor
+		ep = r.endpointManager.LookupIP(addr)
 		if ep != nil {
-			id = ep.GetIdentity()
+			info.ID = ep.GetID()
+			secid, _ := ep.GetSecurityIdentity()
+			if secid != nil {
+				info.Identity = uint64(secid.ID)
+				info.Labels = secid.LabelArray
+			}
 		} else if addr.IsValid() {
 			ID, exists := r.ipcache.LookupByIP(addr.String())
 			if exists {
-				id = ID.ID
+				info.Identity = uint64(ID.ID)
 			}
 		}
 		// Default to WORLD if still unknown
-		if id == 0 {
-			id = identity.GetWorldIdentityFromIP(addr)
+		if info.Identity == 0 {
+			info.Identity = uint64(identity.GetWorldIdentityFromIP(addr))
 		}
 	}
-	info.Identity = uint64(id)
-	identity := r.identityAllocator.LookupIdentityByID(context.TODO(), id)
-	if identity != nil {
-		info.Labels = identity.Labels.GetModel()
+
+	if info.Labels == nil {
+		identity := r.identityAllocator.LookupIdentityByID(context.TODO(), identity.NumericIdentity(info.Identity))
```

Comment:
I understand is was there before the patch, but are we ok with `context.TODO()` here? Shouldn't we at least use a timeout?

Reply:
We should probably add a timeout, yes. This logging is in the critical path for the DNS proxy.

Fortunately, this goes through the identity cache, so the chances of timeout are very, very low. I'll add it as a separate commit.